CHANGELOG.md

```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [5.23.0] - TBD
+
+### Updated
+- Specify Python 3.11 in virtual environment for development and pin `pytest` at version 8.1.1 to match
+
```

Comment:
@LiamConnors which version we used for the last plotly.py release?

Reply:
Also wondering perhaps we wanna mention the version in release docs below? https://github.com/plotly/plotly.py/blob/master/release.md

Reply:
The release build uses 3.9 and the docs build also currently uses 3.9, though I have a PR to upgrade the docs build to 3.10

Reply:
According to https://docs.python.org/3/whatsnew/3.11.html
3.11 seems to run faster when compared to 3.10.

Do you think we could/should upgrade those to 3.10 or 3.11?

Reply:
would you like me to check that the newly-pinned version of pytest (8.1.1) works with Python 3.9 and Python 3.10 as well as Python 3.11? `contributing.md` explicitly mentions Python 3.6 and Python 3.9 in conjunction with `tox`, which I haven't been using so far, but I could set that up and test it later today or tomorrow. RSVP.

Reply:
That would be great. 
Yeah, I guess the `tox` stuff is potentially out of date now. I haven't tried those steps https://github.com/plotly/plotly.py/blob/master/contributing.md#running-tests-with-tox but I'll try with a few different Python versions.